File: scripts/build-resume.py
"""Generate Gregory Uku's résumé as a single-page PDF.

Output: public/resume.pdf

Sources of truth (so the résumé stays honest about what's on the site):
  src/data/projects.js     — projects
  src/data/skills.js       — skills
  src/data/experience.js   — work
  src/data/education.js    — Trent program + courses

Re-runnable: edit the data files, run `python scripts/build-resume.py`, commit.

Design: ATS-friendly text PDF, single column, Helvetica family, ember accent
on the name + section headers so it matches the portfolio's brand without
sacrificing parseability.
"""

from pathlib import Path

from reportlab.lib.colors import HexColor
from reportlab.lib.pagesizes import LETTER
from reportlab.pdfgen import canvas

ROOT = Path(__file__).resolve().parent.parent
OUT = ROOT / "public" / "resume.pdf"

EMBER = HexColor("#ff5b22")
BONE = HexColor("#1a1a1f")
DIM = HexColor("#5a5a63")
LINE = HexColor("#d8d4cb")
TIDE = HexColor("#1d4ed8")

PAGE_W, PAGE_H = LETTER
MARGIN_X = 50
MARGIN_T = 50


def main() -> None:
    c = canvas.Canvas(str(OUT), pagesize=LETTER)
    c.setTitle("Gregory Uku — Résumé")
    c.setAuthor("Gregory Uku")
    c.setSubject("Software Engineering · Resume 2026")
    c.setCreator("portfolio/scripts/build-resume.py")

    y = PAGE_H - MARGIN_T

    # ── Header ──────────────────────────────────────────────────────────
    c.setFillColor(BONE)
    c.setFont("Helvetica-Bold", 28)
    c.drawString(MARGIN_X, y, "Gregory Uku")

    y -= 16
    c.setFont("Helvetica", 11)
    c.setFillColor(EMBER)
    c.drawString(MARGIN_X, y, "Software Engineering student · Builder")
    c.setFillColor(DIM)
    c.setFont("Helvetica-Oblique", 10)
    open_x = MARGIN_X + c.stringWidth(
        "Software Engineering student · Builder  ", "Helvetica", 11
    )
    c.drawString(open_x, y, "· Open to internships, freelance, collaboration")

    y -= 14
    c.setFont("Helvetica", 10)
    c.setFillColor(DIM)
    contact = (
        "ukugregory@gmail.com  ·  github.com/Dekryon  ·  "
        "linkedin.com/in/gregory-uku-8b632724b  ·  Peterborough, ON"
    )
    c.drawString(MARGIN_X, y, contact)

    y -= 10
    c.setStrokeColor(EMBER)
    c.setLineWidth(1.2)
    c.line(MARGIN_X, y, PAGE_W - MARGIN_X, y)

    # ── Summary ─────────────────────────────────────────────────────────
    y -= 20
    c.setFillColor(BONE)
    c.setFont("Helvetica", 10)
    summary = (
        "I design and ship real-world software, web apps, and AI-powered tools. "
        "Comfortable across the stack — React + Tailwind on the front, Node + "
        "TypeScript + Anthropic / OpenAI APIs on the back, Supabase / Vercel for "
        "data and deploy. Multiple shipped products: ai-saas-factory, ordo-portal "
        "(AI drive-thru), Quality Auto Signatures, and this portfolio. I learn "
        "fast and ship what I learn."
    )
    y = wrap_text(c, summary, MARGIN_X, y, PAGE_W - 2 * MARGIN_X, 11, "Helvetica", 10)

    # ── Education ───────────────────────────────────────────────────────
    y = section_header(c, "Education", y)
    c.setFillColor(BONE)
    c.setFont("Helvetica-Bold", 11)
    c.drawString(MARGIN_X, y, "Trent University")
    c.setFont("Helvetica", 10)
    c.setFillColor(DIM)
    c.drawRightString(PAGE_W - MARGIN_X, y, "Peterborough, ON · 2023 — Present")
    y -= 13
    c.setFillColor(BONE)
    c.setFont("Helvetica-Oblique", 10)
    c.drawString(MARGIN_X, y, "B.Sc. · Computer Science / Software Engineering")
    y -= 14
    c.setFillColor(DIM)
    c.setFont("Helvetica", 9)
    courses = (
        "Relevant coursework · Data Management Systems, Software Design & Analysis, "
        "Computer Architecture, Web Development, Linear Algebra, Statistics."
    )
    y = wrap_text(c, courses, MARGIN_X, y, PAGE_W - 2 * MARGIN_X, 10, "Helvetica", 9)

    # ── Projects ────────────────────────────────────────────────────────
    y = section_header(c, "Selected Projects", y)
    projects = [
        (
            "AI SaaS Factory",
            "github.com/Dekryon/ai-saas-factory · ai-saas-factory-ten.vercel.app",
            "Next.js · TypeScript · Anthropic API · E2B · Supabase · Replicate",
            "Autonomous AI agent mesh that turns one-sentence ideas into a shipped "
            "SaaS — orchestrator fans out to builder, marketing, and security "
            "agents in parallel; results land in a single review-and-deploy dashboard.",
        ),
        (
            "Ordo · AI Drive-Thru",
            "github.com/Dekryon/drive-thru-ai · ordo-portal-liart.vercel.app",
            "React · Node.js · OpenAI API · Deepgram · WebSockets · Tailwind",
            "Voice-first drive-thru assistant: speech-to-intent pipeline with "
            "confidence-aware confirmation flow, operator override dashboard, and "
            "per-location menu config. Live pilot in Canada.",
        ),
        (
            "Quality Auto Signatures",
            "qualityautosignatures.com",
            "HTML · CSS · JavaScript · Netlify",
            "Trust-first website for a Nigerian vehicle business — inventory grid, "
            "vehicle detail pages, WhatsApp + email contact flow, mobile-first "
            "layout, SEO and OG metadata. Shipped and in production.",
        ),
        (
            "This Portfolio",
            "github.com/Dekryon/portfolio",
            "React · Vite · Three.js (R3F) · Lenis · Framer Motion · Tailwind",
            "Cinematic dev portfolio: real WebGL hero (wireframe sphere, ember orbit, "
            "bloom + chromatic aberration), Lenis smooth scroll, sticky-scroll project "
            "sections, Instrument Serif display type. WCAG 2.1 AA pass.",
        ),
    ]
    for title, link, stack, body in projects:
        y = project_block(c, title, link, stack, body, y)
        if y < 200:
            break

    # ── Experience ──────────────────────────────────────────────────────
    y = section_header(c, "Experience", y)
    experiences = [
        (
            "Student Housing · Customer Service",
            "Trent University · 2024 — Present",
            "Frontline support for residents — owning the response loop on requests, "
            "troubleshooting, and escalations. Logged + routed without dropping balls.",
        ),
        (
            "Freelance · Laptop Repair & IT Help",
            "Self-employed · 2022 — Present",
            "Hands-on troubleshooting across Windows, macOS, and Linux. Hardware, "
            "software, malware, data recovery, machine migration. Wrote how-to notes "
            "so clients could self-serve later.",
        ),
        (
            "Choir Director",
            "Community Choir · 2022 — Present",
            "Lead weekly rehearsals for 15+ members, arrange pieces, assign parts, "
            "coordinate live event logistics end-to-end.",
        ),
        (
            "Sales Associate",
            "Retail · 2023 — 2024",
            "Customer-facing sales. Listened fast, asked the right questions, hit "
            "consistent targets through honest recommendations.",
        ),
    ]
    for title, where, body in experiences:
        y = experience_block(c, title, where, body, y)
        if y < 130:
            break

    # ── Skills ──────────────────────────────────────────────────────────
    y = section_header(c, "Skills", y)
    skills_lines = [
        ("Languages", "HTML, CSS, JavaScript, C#, TypeScript, Python, SQL"),
        ("Frontend", "React, Tailwind CSS, Framer Motion, Responsive design, UI design"),
        ("Backend / data", "Node.js, Supabase, REST APIs, JSON, Fetch / Axios"),
        ("AI tooling", "Anthropic SDK, OpenAI API, E2B, Replicate, AI-assisted dev"),
        ("Tools", "Git, GitHub, Vite, Netlify, Vercel, VS Code"),
    ]
    for label, body in skills_lines:
        c.setFillColor(EMBER)
        c.setFont("Helvetica-Bold", 9)
        c.drawString(MARGIN_X, y, label.upper())
        c.setFillColor(BONE)
        c.setFont("Helvetica", 9)
        c.drawString(MARGIN_X + 90, y, body)
        y -= 11

    # ── Footer line ─────────────────────────────────────────────────────
    c.setStrokeColor(LINE)
    c.setLineWidth(0.4)
    c.line(MARGIN_X, 42, PAGE_W - MARGIN_X, 42)
    c.setFillColor(DIM)
    c.setFont("Helvetica", 8)
    c.drawString(MARGIN_X, 32, "Generated from portfolio/src/data — re-run scripts/build-resume.py to refresh")
    c.drawRightString(PAGE_W - MARGIN_X, 32, "Gregory Uku · 2026")

    c.save()
    size_kb = OUT.stat().st_size // 1024
    print(f"saved {OUT.relative_to(ROOT)} ({size_kb} KB)")


def section_header(c: canvas.Canvas, label: str, y: float) -> float:
    y -= 18
    c.setFillColor(EMBER)
    c.setFont("Helvetica-Bold", 10)
    c.drawString(MARGIN_X, y, label.upper())
    text_width = c.stringWidth(label.upper(), "Helvetica-Bold", 10)
    c.setStrokeColor(LINE)
    c.setLineWidth(0.5)
    c.line(MARGIN_X + text_width + 8, y + 3, PAGE_W - MARGIN_X, y + 3)
    return y - 13


def project_block(
    c: canvas.Canvas, title: str, link: str, stack: str, body: str, y: float
) -> float:
    c.setFillColor(BONE)
    c.setFont("Helvetica-Bold", 11)
    c.drawString(MARGIN_X, y, title)
    c.setFont("Helvetica", 9)
    c.setFillColor(TIDE)
    c.drawRightString(PAGE_W - MARGIN_X, y, link)
    y -= 12
    c.setFillColor(DIM)
    c.setFont("Helvetica-Oblique", 9)
    c.drawString(MARGIN_X, y, stack)
    y -= 12
    c.setFillColor(BONE)
    y = wrap_text(c, body, MARGIN_X, y, PAGE_W - 2 * MARGIN_X, 10, "Helvetica", 9)
    return y - 5


def experience_block(
    c: canvas.Canvas, title: str, where: str, body: str, y: float
) -> float:
    c.setFillColor(BONE)
    c.setFont("Helvetica-Bold", 11)
    c.drawString(MARGIN_X, y, title)
    c.setFont("Helvetica", 9)
    c.setFillColor(DIM)
    c.drawRightString(PAGE_W - MARGIN_X, y, where)
    y -= 12
    c.setFillColor(BONE)
    y = wrap_text(c, body, MARGIN_X, y, PAGE_W - 2 * MARGIN_X, 10, "Helvetica", 9)
    return y - 4


def wrap_text(
    c: canvas.Canvas,
    text: str,
    x: float,
    y: float,
    max_width: float,
    line_height: float,
    font: str,
    size: int,
) -> float:
    c.setFont(font, size)
    words = text.split()
    line = ""
    for word in words:
        candidate = (line + " " + word).strip()
        if c.stringWidth(candidate, font, size) > max_width:
            c.drawString(x, y, line)
            y -= line_height
            line = word
        else:
            line = candidate
    if line:
        c.drawString(x, y, line)
        y -= line_height
    return y


if __name__ == "__main__":
    main()

File: scripts/build-resume.mjs
// Generate Gregory Uku's résumé as a single-page PDF.
//
// Output: public/resume.pdf
//
// Sources of truth (kept in lock-step with the live site):
//   src/data/projects.js     — projects
//   src/data/experience.js   — work
//   src/data/education.js    — Trent program + courses
//
// Re-runnable: `node scripts/build-resume.mjs` (or `npm run resume`).
// HTML + Playwright print → real typography, single-column ATS-friendly.

import { chromium } from 'playwright'
import { fileURLToPath } from 'node:url'
import { dirname, join } from 'node:path'
import { writeFile } from 'node:fs/promises'

import { PROJECTS } from '../src/data/projects.js'
import { EXPERIENCE } from '../src/data/experience.js'
import { EDUCATION } from '../src/data/education.js'

const __dirname = dirname(fileURLToPath(import.meta.url))
const OUT_PDF = join(__dirname, '..', 'public', 'resume.pdf')

// ── Résumé-specific copy (the parts not in the site data) ───────────────

const NAME = 'Gregory Uku'
const ROLE_LINE_1 = 'Founder & CEO · Ordo'
const ROLE_LINE_2 = 'Software Engineering Student · Trent University'
const OPEN_TO = 'Open to investor intros · partnerships · internships'

const CONTACT = [
  { label: 'ukugregory@gmail.com', href: 'mailto:ukugregory@gmail.com' },
  { label: 'gregoryuku.com', href: 'https://gregoryuku.com' },
  { label: 'github.com/Dekryon', href: 'https://github.com/Dekryon' },
  { label: 'linkedin.com/in/gregory-uku', href: 'https://www.linkedin.com/in/gregory-uku-8b632724b' },
  { label: 'Peterborough, ON', href: null }
]

const SUMMARY =
  'Founder building AI products with real customers. Currently CEO of Ordo (AI phone receptionist · live pilot in Canada) and founder of Signal (AI trading intelligence). Three years of shipping · client websites, multi-agent systems, a portfolio of live URLs. Software Engineering at Trent while building.'

// ── Project copy is sourced from src/data/projects.js but tightened for
// the résumé's tighter width. Title, role label, demo URL, stack, one-line
// pitch. No vendor names. Only public/working URLs.
const RESUME_PROJECTS = [
  {
    id: 'ordo',
    role: 'Founder & CEO',
    period: '2026 · Present',
    pitch:
      'AI phone receptionist that answers every call · books, orders, FAQs, messages · in your brand voice, 24/7. Three vertical doors (restaurants, hotels, medical). $149 per line per month. Live pilot in Canada with 94% of orders handled end to end.',
    tech: 'Next.js · Node · Voice AI · WebSockets · Tailwind'
  },
  {
    id: 'signal',
    role: 'Founder & Engineer',
    period: '2026 · Present',
    pitch:
      'AI trading intelligence. Five specialized agents (Insider, Options, News, Pre-Market, Macro) brief a synthesizer that writes one BUY / SELL / HOLD / AVOID call per ticker with stop loss, take profit, and position size. Paper-traded until win rate is proven.',
    tech: 'Next.js · TypeScript · Postgres · Real-time data feeds'
  },
  {
    id: 'ai-saas-factory',
    role: 'Solo · Design + Engineering',
    period: '2026',
    pitch:
      'Autonomous agent mesh that turns a one-line product brief into a deployable SaaS. Orchestrator fans out to builder, marketing, and security agents in parallel; results land in one review-and-deploy dashboard. Open beta.',
    tech: 'Next.js · TypeScript · Code sandboxes · Postgres'
  },
  {
    id: 'quality-auto',
    role: 'Solo · Design + Build · Client work',
    period: '2025',
    pitch:
      'Trust-first website for a Nigerian vehicle business. Inventory grid, vehicle detail pages, WhatsApp + email contact flow, SEO and Open Graph metadata. Shipped to production.',
    tech: 'HTML · CSS · JavaScript · Netlify'
  }
]

// Cross-reference with PROJECTS so demo URLs stay accurate when the data changes.
const PROJECT_URLS = Object.fromEntries(PROJECTS.map((p) => [p.id, { demo: p.demo, title: p.title, subtitle: p.subtitle, github: p.github }]))

const RESUME_EXPERIENCE = [
  {
    role: 'Founder & CEO',
    org: 'Ordo',
    period: '2026 · Present',
    body: 'Building AI phone receptionist live-piloted in Canada. Designed and shipped the marketing landing (useordo.org), operator portal (app.useordo.org), and voice agent. Same brain, three vertical doors. Carrying product, engineering, and early sales single-handed.'
  },
  {
    role: 'Student Housing · Customer Service',
    org: 'Trent University',
    period: '2024 · Present',
    body: 'Frontline support for residents · owned the response loop on requests, troubleshooting, and escalations. Logged and routed without dropping balls, built rapport during high-stress periods.'
  },
  {
    role: 'Freelance · Laptop Repair & IT Help',
    org: 'Self-employed',
    period: '2022 · Present',
    body: 'Hands-on troubleshooting across Windows, macOS, and Linux. Hardware, software, malware, data recovery, machine migration. Wrote how-to notes so clients could self-serve later.'
  },
  {
    role: 'Choir Director',
    org: 'Community Choir',
    period: '2022 · Present',
    body: 'Lead weekly rehearsals for 15+ members, arrange pieces, assign parts, coordinate live event logistics end-to-end.'
  }
]

const SKILLS = [
  { label: 'Languages', body: 'TypeScript, JavaScript, Python, C#, SQL, HTML, CSS' },
  { label: 'Frontend', body: 'React, Next.js, Tailwind, Framer Motion, Three.js (R3F), Responsive + accessibility' },
  { label: 'Backend / Data', body: 'Node.js, Postgres, REST APIs, WebSockets, real-time pipelines, edge functions' },
  { label: 'AI Engineering', body: 'Multi-agent orchestration, voice AI, prompt + tool design, retrieval, evals' },
  { label: 'Tooling', body: 'Git, GitHub, Vite, Vercel, Netlify, Playwright, VS Code' }
]

// ── HTML template ────────────────────────────────────────────────────────

function escape(s) {
  return String(s).replace(/[&<>"']/g, (c) => ({ '&': '&amp;', '<': '&lt;', '>': '&gt;', '"': '&quot;', "'": '&#39;' }[c]))
}

function projectsHtml() {
  return RESUME_PROJECTS.map((p) => {
    const live = PROJECT_URLS[p.id]
    const demo = live?.demo
    const title = live?.title || p.id
    const subtitle = live?.subtitle
    const demoLabel = demo ? demo.replace(/^https?:\/\//, '').replace(/\/$/, '') : ''
    return `
      <div class="row">
        <div class="row-head">
          <div>
            <div class="row-title">${escape(title)}${subtitle ? ` <span class="row-sub">· ${escape(subtitle)}</span>` : ''}</div>
            <div class="row-meta">${escape(p.role)}</div>
          </div>
          <div class="row-right">
            ${demo ? `<a href="${escape(demo)}">${escape(demoLabel)}</a>` : ''}
            <div class="row-dim">${escape(p.period)}</div>
          </div>
        </div>
        <p class="row-body">${escape(p.pitch)}</p>
        <p class="row-tech">${escape(p.tech)}</p>
      </div>`
  }).join('')
}

function experienceHtml() {
  return RESUME_EXPERIENCE.map((e) => `
    <div class="row tight">
      <div class="row-head">
        <div>
          <div class="row-title">${escape(e.role)} <span class="row-sub">· ${escape(e.org)}</span></div>
        </div>
        <div class="row-right">
          <div class="row-dim">${escape(e.period)}</div>
        </div>
      </div>
      <p class="row-body">${escape(e.body)}</p>
    </div>`).join('')
}

function skillsHtml() {
  return SKILLS.map((s) => `
    <div class="skill-row">
      <div class="skill-label">${escape(s.label)}</div>
      <div class="skill-body">${escape(s.body)}</div>
    </div>`).join('')
}

function educationHtml() {
  return `
    <div class="row tight">
      <div class="row-head">
        <div>
          <div class="row-title">${escape(EDUCATION.school)} <span class="row-sub">· ${escape(EDUCATION.program)}</span></div>
        </div>
        <div class="row-right">
          <div class="row-dim">${escape(EDUCATION.location.split(',').slice(0, 2).join(', '))} · ${escape(EDUCATION.period)}</div>
        </div>
      </div>
      <p class="row-body">Relevant coursework · ${EDUCATION.courses.map((c) => escape(c.name)).join(', ')}.</p>
    </div>`
}

function contactHtml() {
  return CONTACT.map((c) => c.href
    ? `<a href="${escape(c.href)}">${escape(c.label)}</a>`
    : `<span>${escape(c.label)}</span>`
  ).join('<span class="contact-sep">·</span>')
}

const HTML = `<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8" />
  <title>${escape(NAME)} · Résumé</title>
  <link rel="preconnect" href="https://fonts.googleapis.com" />
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700&family=Instrument+Serif:ital@0;1&display=swap" />
  <style>
    @page { size: Letter; margin: 0; }
    :root {
      --ink: #1a1a1f;
      --dim: #5a5a63;
      --muted: #8a8a93;
      --line: #e0dcd2;
      --ember: #ff5b22;
      --serif: 'Instrument Serif', Georgia, serif;
      --sans: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', sans-serif;
    }
    * { box-sizing: border-box; }
    html, body { margin: 0; padding: 0; }
    body {
      font-family: var(--sans);
      color: var(--ink);
      font-size: 9pt;
      line-height: 1.4;
      background: #fff;
      -webkit-print-color-adjust: exact;
      print-color-adjust: exact;
    }
    .page {
      width: 8.5in;
      min-height: 11in;
      padding: 0.4in 0.55in 0.28in 0.55in;
    }

    /* Header */
    .header { display: flex; align-items: flex-start; justify-content: space-between; gap: 24px; }
    .name {
      font-family: var(--serif);
      font-size: 38pt;
      line-height: 0.95;
      letter-spacing: -0.02em;
      margin: 0;
    }
    .name em { font-style: italic; color: var(--ember); }
    .role { font-size: 10pt; color: var(--dim); margin-top: 6px; }
    .role .role-1 { color: var(--ember); font-weight: 600; }
    .role .sep { color: var(--muted); margin: 0 6px; }
    .open-to { font-size: 8.5pt; color: var(--muted); font-style: italic; margin-top: 4px; }

    .contact {
      text-align: right;
      font-size: 8.5pt;
      color: var(--dim);
      max-width: 3.4in;
      line-height: 1.7;
    }
    .contact a { color: var(--dim); text-decoration: none; border-bottom: 1px solid transparent; }
    .contact a:hover { border-bottom-color: var(--ember); }
    .contact .contact-sep { margin: 0 6px; color: var(--muted); }

    .accent-rule {
      height: 2px;
      background: var(--ember);
      margin: 10px 0 10px 0;
      width: 100%;
    }

    /* Summary */
    .summary {
      font-size: 9.5pt;
      line-height: 1.45;
      color: var(--ink);
      margin: 0 0 2px 0;
      max-width: 7.2in;
    }

    /* Sections */
    .section { margin-top: 9px; }
    .section-head {
      display: flex; align-items: baseline; gap: 12px;
      margin-bottom: 5px;
    }
    .section-head .label {
      font-family: var(--sans);
      font-size: 8.5pt;
      letter-spacing: 0.18em;
      text-transform: uppercase;
      color: var(--ember);
      font-weight: 600;
    }
    .section-head .rule { flex: 1; height: 1px; background: var(--line); }

    .row { margin-bottom: 6px; }
    .row.tight { margin-bottom: 4px; }
    .row:last-child { margin-bottom: 0; }
    .row-head {
      display: flex; justify-content: space-between; align-items: baseline;
      gap: 16px;
    }
    .row-title {
      font-weight: 600;
      font-size: 10.5pt;
      color: var(--ink);
    }
    .row-title .row-sub {
      font-weight: 400;
      font-style: italic;
      color: var(--dim);
      font-family: var(--serif);
      font-size: 11pt;
    }
    .row-meta { font-size: 8.5pt; color: var(--dim); margin-top: 1px; }
    .row-right { text-align: right; }
    .row-right a {
      color: var(--ember);
      text-decoration: none;
      font-size: 8.5pt;
      font-weight: 500;
    }
    .row-dim { font-size: 8pt; color: var(--muted); margin-top: 1px; letter-spacing: 0.03em; }
    .row-body {
      font-size: 9pt;
      color: var(--ink);
      margin: 2px 0 1px 0;
      max-width: 7.2in;
      line-height: 1.4;
    }
    .row-tech {
      font-size: 8pt;
      color: var(--dim);
      font-style: italic;
      margin: 0;
      max-width: 7.2in;
    }

    /* Skills */
    .skill-row {
      display: grid;
      grid-template-columns: 1.1in 1fr;
      gap: 8px;
      padding: 2px 0;
      align-items: baseline;
    }
    .skill-label {
      font-size: 8pt;
      letter-spacing: 0.15em;
      text-transform: uppercase;
      color: var(--ember);
      font-weight: 600;
    }
    .skill-body { font-size: 9.5pt; color: var(--ink); }

    /* Footer */
    .footer {
      margin-top: 10px;
      padding-top: 5px;
      border-top: 1px solid var(--line);
      display: flex; justify-content: space-between;
      font-size: 7.5pt;
      color: var(--muted);
      letter-spacing: 0.08em;
    }
  </style>
</head>
<body>
  <div class="page">
    <div class="header">
      <div>
        <h1 class="name">Gregory <em>Uku</em></h1>
        <div class="role">
          <span class="role-1">${escape(ROLE_LINE_1)}</span>
          <span class="sep">·</span>
          <span>${escape(ROLE_LINE_2)}</span>
        </div>
        <div class="open-to">${escape(OPEN_TO)}</div>
      </div>
      <div class="contact">${contactHtml()}</div>
    </div>

    <div class="accent-rule"></div>

    <p class="summary">${escape(SUMMARY)}</p>

    <section class="section">
      <div class="section-head"><span class="label">Selected Work</span><span class="rule"></span></div>
      ${projectsHtml()}
    </section>

    <section class="section">
      <div class="section-head"><span class="label">Experience</span><span class="rule"></span></div>
      ${experienceHtml()}
    </section>

    <section class="section">
      <div class="section-head"><span class="label">Skills</span><span class="rule"></span></div>
      ${skillsHtml()}
    </section>

    <section class="section">
      <div class="section-head"><span class="label">Education</span><span class="rule"></span></div>
      ${educationHtml()}
    </section>
  </div>
</body>
</html>
`

async function main() {
  const browser = await chromium.launch()
  const context = await browser.newContext({ viewport: { width: 816, height: 1056 } })
  const page = await context.newPage()
  await page.setContent(HTML, { waitUntil: 'networkidle' })
  await page.emulateMedia({ media: 'print' })
  await page.pdf({
    path: OUT_PDF,
    width: '8.5in',
    height: '11in',
    margin: { top: 0, right: 0, bottom: 0, left: 0 },
    printBackground: true,
    preferCSSPageSize: true
  })
  await browser.close()
  // Echo the rendered HTML alongside in case we want to host it too.
  const htmlOut = OUT_PDF.replace(/\.pdf$/, '.html')
  await writeFile(htmlOut, HTML, 'utf-8')
  console.log(`saved public/resume.pdf and public/resume.html`)
}

main().catch((err) => { console.error(err); process.exit(1) })
